application/comit_node/src/comit_i_routes.rs

```diff
@@ -0,0 +1,98 @@
+use comit_i::Asset;
+use http::{uri::PathAndQuery, Response, StatusCode};
+use mime_guess::Mime;
+use std::{
+    borrow::Cow,
+    error::Error as StdError,
+    fmt::{self, Display},
+    str::FromStr,
+};
+use warp::{filters::BoxedFilter, path::Tail, Filter, Rejection, Reply};
+
+#[derive(Copy, Clone, Debug)]
+enum Error {
+    IndexHtmlMissing,
+    PathConversionFail,
+}
+
+#[derive(Serialize)]
+struct ErrorMessage {
+    code: u16,
+    message: String,
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.description())
+    }
+}
+
+impl From<Error> for Rejection {
+    fn from(err: Error) -> Self {
+        warp::reject::custom(err)
+    }
+}
+
+impl StdError for Error {
+    fn description(&self) -> &str {
+        match self {
+            Error::IndexHtmlMissing => "index.html file not found",
+            Error::PathConversionFail => "Conversion of the path to PathAndQuery failed",
+        }
+    }
+
+    fn cause(&self) -> Option<&dyn StdError> {
+        None
+    }
+}
+
+pub fn create() -> BoxedFilter<(impl Reply,)> {
```

Comment:
:wrench: I liked it better when this was part of the `comit_i` crate. Why did you remove it?

Reply:
What was in the comit_i crate was just an example to use rust-embed with warp. I have now moved it directly to rust-embed repo.

To me, the comit_i crate just provides comit_i. The lib consumer should have the choice on how to serve it (warp, rocket, etc).
I did not want to make this choice for the consumer that is why it is in comit_node crate.

Reply:
Hmm, comit_i by itself is provided in its own repository. Everything that is in the `comit_i` crate can be trivially built by everyone (no real gain of re-use here).

For now though, only the functionality matters :)